src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2047,7 +2048,7 @@ JoinRuleApplyFunction(JoinRuleType ruleType)
  */
 static MultiNode *
 ApplyReferenceJoin(MultiNode *leftNode, MultiNode *rightNode,
```

Comment:
Could be a personal view, so feel free to ignore. I'd have used `partitionColumnList` instead of `partitionColumns ` as the former is little more expressive, and we typically follow that convention in the code